### ISSUE TITLE
Feature Request: View Switch or similar for mobile handset

### ISSUE THREAD

**Issue by 000exploit** [xou816/spot#471]
Currently, I'm using spot also on a mobile device named as Lenovo A6010, and I'm not happy with current UX, that is really inconvenient on small and big screens(of course on mobile devices). Always rely on `GtkListBox` for every action isn't good for daily usage, but perhaps the solution for this problem may be View Switch(or top/dockbar as it referenced in some design docs), that can separate "Home" page(see #468), search and "Library" as in official mobile app. There are some reference in screenshots from official mobile app:
<details><summary>Dockbar example</summary>

![Dockbar](https://user-images.githubusercontent.com/28919772/154355761-8e385333-f7b4-45eb-9ded-f2de9db6407e.png)</details>
And
<details><summary>Library menu example</summary>

![Library](https://user-images.githubusercontent.com/28919772/154355985-0f0b13ee-d364-46a0-a70e-9443b7aac474.png)</details>

As I know, GTK and libadwaita allows changing elements in runtime(i.e. resizing window), so it should be possible to implement without changing "desktop" interface.

**Comment by jannuary** [comment]
This makes sense to me, as long as it's technically feasible without too much jank - Software abandoned the desktop sidebar, mobile view switcher paradigm, for example: [Reconsider sidebar for 41?](https://gitlab.gnome.org/GNOME/gnome-software/-/issues/1278)

**Comment by 000exploit** [comment]
> Software abandoned the desktop sidebar, mobile view switcher paradigm

Yes, I saw how worst it can be [here](https://gitlab.gnome.org/GNOME/gnome-software/-/issues/1278#note_1194521) after resize, but they chose View Switcher on all devices that works in their case, but not here. App will be unusable on desktop if there will be only View Switcher and with sidebar it will require additional work. I don't agree to leave everything as it is, because app already adaptive and can be used on the phone.

**Comment by xou816** [comment]
This makes sense -- would have to rethink quite a few things if we get rid of the sidebar though

**Comment by 000exploit** [comment]
It may be solvable after the release of libadwaita 1.3, adaptive states will be added there (they should be).
